petri/cosmos/chain/chain.go

```diff
@@ -788,11 +813,214 @@ func (c *Chain) executeGenesisOperations(ctx context.Context, firstValidator pet
 		return fmt.Errorf("final genesis script failed (exit code %d): %s, stdout: %s", exitCode, stderr, stdout)
 	}
 
+	if accountOpts.numAdditionalAccounts > 0 {
+		if accountOpts.baseMnemonic == "" {
+			return fmt.Errorf("base-mnemonic is required when additional accounts > 0")
+		}
+		accounts, err := buildAccounts(walletCfg, accountOpts.baseMnemonic, len(c.Validators)+1, accountOpts.numAdditionalAccounts)
+		if err != nil {
+			return fmt.Errorf("failed to build additional accounts: %w", err)
+		}
+		balances := buildBalances(accounts, genesisAmounts)
+		eg := new(errgroup.Group)
+		for _, v := range c.Validators {
+			eg.Go(func() error {
+				genesisBz, err := v.GenesisFileContent(ctx)
+				if err != nil {
+					return fmt.Errorf("failed to get genesis file: %w", err)
+				}
+				data := make(map[string]any)
+				err = json.Unmarshal(genesisBz, &data)
+				if err != nil {
+					return err
+				}
+
+				data, err = UpdateGenesisAccounts(accounts, data)
+				if err != nil {
+					return fmt.Errorf("failed to update genesis accounts: %w", err)
+				}
+
+				data, err = UpdateGenesisBalances(balances, data)
+				if err != nil {
+					return fmt.Errorf("failed to update genesis balances: %w", err)
+				}
+
+				updatedGenesisBz, err := json.Marshal(data)
+				if err != nil {
+					return err
+				}
+
+				err = v.OverwriteGenesisFile(ctx, updatedGenesisBz)
+				if err != nil {
+					return fmt.Errorf("failed to overwrite genesis file for validator: %w", err)
+				}
+				return nil
+			})
+		}
+		if err := eg.Wait(); err != nil {
+			return fmt.Errorf("failed to add additional accounts: %w", err)
+		}
+	}
```

Comment:
I think we discussed that the genesis bytes are currently generated in a single place and then written to all of the validators, whereas here you're concurrently executing the genesis operations on each validator's file.

We should probably refactor this in the future since it seems unnecessary to do this in multiple places.